### PR TITLE
refactor: rename `modules` project config entry to `scan`

### LINKED ISSUE #3512: Rename Project config `modules.*` to `scan.*`
Rename Project config `modules.*` to `scan.*`. Allow the alias with a deprecation notice. Include this in `migrate` command.

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3512

**Special notes for your reviewer**:

### CODE REVIEW

File: core/src/config/project.ts [hunk 1]
```diff
@@ -187,10 +187,18 @@ export interface ProjectConfig {
   dotIgnoreFile: string
   dotIgnoreFiles?: string[]
   environments: EnvironmentConfig[]
+  /**
+   * Deprecated. Use {@link #scan} instead.
+   * @deprecated remove in 0.14
+   */
   modules?: {
```
Comment (edvald) [inline core/src/config/project.ts:194]: I think we should just remove this from the internal interface.

File: core/src/config/base.ts [hunk 2]
```diff
@@ -290,11 +290,56 @@ export function prepareProjectResource(log: Log, spec: any): ProjectResource {
...
+    log,
+    "Project configuration field `modules` is deprecated in 0.13 and will be removed in 0.14. Please use the `scan` field instead."
+  )
+
+  if (!projectSpec.scan) {
+    projectSpec.scan = {}
+  }
+
+  if (projectSpec["modules"].include) {
+    const oldSyntaxInclusions = projectSpec["modules"].include || []
```
Comment (edvald) [inline core/src/config/base.ts:315]: Why not simply use the rename option in the joi schema? You can catch and emit the usage of the old field before validating.

Comment (vvagaytsev) [reply core/src/config/base.ts:315]: That was my initial plan. Should we deny using both fields simultaneously?

Comment (vvagaytsev) [reply core/src/config/base.ts]: Simplified this in https://github.com/garden-io/garden/pull/4036/commits/2966c7f86ef5b6f6c7f2c7c6fc428b9c808ba6e0

**Pull request merged**